src/app/auth/auth-direct-login/auth-direct-login.component.ts

```diff
@@ -55,7 +55,7 @@ export class AuthDirectLoginComponent implements OnInit {
   /**
    * Redirect user to a specific page if data is passed in, otherwise redirect to program switcher page
    */
-  private async _redirect(): Promise<boolean> {
+  private async _redirect(notRegister?: boolean) {
```

Comment:
I think it's better to name this variable something like `redirectLater` or `delayedRedirect`. So that from the name we know we are going to redirect the user later.
Otherwise, we will forget what does "not register" mean to the function "redirect".